Resources/Scripts/search_filter_suggestions.js

```diff
@@ -0,0 +1,102 @@
+    })
+
+    document.addEventListener('turbo:load', () => {
+      const searchSectionElement = document.querySelector('[data-filter-suggestions]')
+      if (!searchSectionElement) return
+      const searchFieldElement = searchSectionElement.querySelector('form input[type=search]')
+      if (!searchFieldElement) return
+
+      // Add the search suggestions below the search field.
+      const filterSuggestionsElement = document.createElement('div')
+      filterSuggestionsElement.classList.add('filter_suggestions')
+      filterSuggestionsElement.innerHTML =
+        'Add filters to narrow search results (<a href="/faq#search-filters">Learn more</a>). For example: '
+      searchSectionElement.appendChild(filterSuggestionsElement)
+
+      SPISearchFilterSuggestions.suggestions.forEach((suggestion) => {
+        const linkElement = document.createElement('a')
+        linkElement.textContent = suggestion.filter
+        linkElement.title = suggestion.description
+        linkElement.dataset.filter = suggestion.filter
+        filterSuggestionsElement.appendChild(linkElement)
+
+        // Top and tail with quotes.
+        // Note: The element *must* be inserted into the DOM for this to work.
+        linkElement.insertAdjacentHTML('beforebegin', '&ldquo;')
+        linkElement.insertAdjacentHTML('afterend', '&rdquo; ')
+
+        linkElement.addEventListener('click', (event) => {
+          event.preventDefault()
+
+          // Grab the filter and parse it out to get the lengths of each side.
+          const separator = ':'
+          const filter = linkElement.dataset.filter
+          const filterElements = filter.split(separator)
+          const valueLength = filterElements.pop().length
+          const fieldLength = filterElements.pop().length
+          const whitespace = ' ' // To separate the suggested filter from the existing search term
+
+          // Append the filter to the existing search term.
+          var currentSearch = searchFieldElement.value.trimEnd()
+          searchFieldElement.value = currentSearch + whitespace + filter
+
+          // Finally, focus the value portion of the suggested filter.
+          const selectionStart = currentSearch.length + fieldLength + separator.length + whitespace.length
+          const selectionEnd = selectionStart + valueLength
+          searchFieldElement.focus()
+          searchFieldElement.setSelectionRange(selectionStart, selectionEnd, 'forward')
+        })
+      })
+    })
+  }
+
+  static formattedFilterDate() {
+    var thirtyDaysAgo = new Date()
+    thirtyDaysAgo.setDate(thirtyDaysAgo.getDate() - 30)
+
+    const year = thirtyDaysAgo.getFullYear()
+    const month = thirtyDaysAgo.getMonth() + 1 // Yes, JavaScript returns months as zero based.
+    const day = thirtyDaysAgo.getDate() // ... but not the day of the month. That's one based.
+
+    return `${year}-${String(month).padStart(2, '0')}-${String(day).padStart(2, '0')}`
+  }
```

Comment:
This is some of the worst code I can remember writing, but JavaScript does not have freeform date formatting without introducing a dependency which I don't want to do. It does, and should continue to work, but oof... 😬

Reply:
Missed opportunity to use `lpad` 😬